app/models/scimitar/complex_types/base.rb

```diff
@@ -75,6 +75,7 @@ def initialize(options={})
       #             attributes of this complex type object.
       #
       def as_json(options={})
+        options = options.dup if options.frozen?
         options[:except] ||= ['errors']
         super.except(options)
```

Comment:
This is Rails 8 making five minute tasks take hours, again. Rails 7 and 8 have felt like the "bad old days" of Rails 2/3/4 era massive changes in APIs and constant dev churn. I can't find many references to frozen-by-default Hashes anywhere inside the higher level Rails 8 changes, yet it's a major change that (as we see here) can cause breakage.

In any case, I've just realised that this code was broken anyway. The "correct" #as_json takes some finding, but [the relevant `#as_json`'s options include `exclude:`](https://api.rubyonrails.org/classes/ActiveModel/Serializers/JSON.html#method-i-as_json). That's what is being manipulated on line 78.

So we now call `super`, without parens so it gets passed any and all parameters including our options. The result is a Hash. We're then calling [Hash#except](https://docs.ruby-lang.org/en/3.3/Hash.html#method-i-except), but that takes the keys to exclude as a list/Array, yet we're providing an options Hash with who-knows-what as the keys. Besides, since the options were already passed in, exclusions already happened.

I'd like to contribute some minor changes to your branch to get things working for Rails 7/8 and have CI pass (for which I think I might be forced to concede auto-test for Ruby 3.2 or later only and drop tests for earlier versions, even though they still work perfectly). For example, this change - and there's one other place something similar isn needed, if tests are to pass - becomes, in my preferred variation after trying a few:

```ruby
      def as_json(options={})
        exclusions = options[:except] || ['errors']
        super(options.merge(except: exclusions))
      end
```

Reply:
@terracatta ...ah, yay, it *can* be done for old Rubies by modifying the Rails constraint by version. It looks like this: https://github.com/RIPAGlobal/scimitar/pull/154